Add EIP: Withdrawal Credential Update Request (#9005)
ethereum/EIPs: EIPS/eip-7800.md

---

> eip: 7800

Reviewer: ```suggestion
eip: 7804
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

title: Withdrawal Credential Update Request
description: Allow validators to update their withdrawal credentials via execution requests
author: lucassaldanha (@lucassaldanha)

> discussions-to: https://ethereum-magicians.org/t/eip-7800-withdrawal-credential-update-request/21514

Reviewer: ```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7804-withdrawal-credential-update-request/21514
```
Updated with assigned EIP number

status: Draft
type: Standards Track
category: Core
created: 2024-10-31
---

## Abstract

This proposal defines a mechanism to allow validators to update their withdrawal
credentials using a new execution request type (0x03). The request allows for
changing the execution address and the withdrawal credential prefix (0x01 or 0x02).

## Motivation

When the ability to update a validator BLS withdrawal credentials to execution
address was introduced in Capella, one of the most common questions was about
allowing the withdrawal credential to be changed in the future.
Either for security (e.g. credential rotation) or to allow for alternative ways
of handling withdrawals (e.g. having a contract address as credentials).

> The main reason for not adding this options was because the complexity of implementing
> this communication channel between the Execution Layer and the Consensus Layer was
> complicated (based on the experience with the Eth1 bridge).

Reviewer 2: ```suggestion
The main reason for not adding this options was because implementing
this communication channel between the Execution Layer and the Consensus Layer was
complex (based on the experience with the Eth1 bridge).
```
I'd suggest rephrasing this

> The main reason for not adding this options was because the complexity of implementing
> this communication channel between the Execution Layer and the Consensus Layer was
> complicated (based on the experience with the Eth1 bridge).

Reviewer 2: ```suggestion
The main reason for not adding this options was because implementing
this communication channel between the Execution Layer and the Consensus Layer is
complex (based on the experience with the Eth1 bridge).
```
I'd suggest rephrasing this
